Tools/LinearModelTraining/LinearModelTraining.swift

```diff
@@ -7,7 +7,7 @@ import MLXNN
 import MLXOptimizers
 import MLXRandom
 
-#if swift(>=6.0)
+#if swift(>=5.10)
```

Comment:
Fixing some other warnings -- we don't necessarily need swift 6, but not all swift 5 understands this.